Lib/test/test_httpservers.py

```diff
@@ -1535,6 +1537,159 @@ def test_server_test_ipv4(self, _):
+
+        for tls_cert_option in tls_cert_options:
+            for tls_key_option in tls_key_options:
+                self.invoke_httpd([tls_cert_option, self.tls_cert, tls_key_option, self.tls_key])
+                mock_func.assert_called_once_with(HandlerClass=self.default_handler, ServerClass=self.default_server,
+                                                  protocol=self.default_protocol, port=self.default_port, bind=self.default_bind,
+                                                  tls_cert=self.tls_cert, tls_key=self.tls_key, tls_password=None)
+                mock_func.reset_mock()
+
+        # Normal: --tls-cert, --tls-key and --tls-password-file
+
+        for tls_cert_option in tls_cert_options:
+            for tls_key_option in tls_key_options:
+                for tls_password_option in tls_password_options:
+                    self.invoke_httpd([tls_cert_option, self.tls_cert, tls_key_option, self.tls_key, tls_password_option, self.tls_password_file])
+
+                    mock_func.assert_called_once_with(HandlerClass=self.default_handler, ServerClass=self.default_server,
+                                                      protocol=self.default_protocol, port=self.default_port, bind=self.default_bind,
+                                                      tls_cert=self.tls_cert, tls_key=self.tls_key, tls_password=self.tls_password)
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-key without --tls-cert
+
+        for tls_key_option in tls_key_options:
+            for tls_cert_option in tls_cert_options:
+                with self.assertRaises(SystemExit):
+                    self.invoke_httpd([tls_key_option, self.tls_key])
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file without --tls-cert
+
+        for tls_password_option in tls_password_options:
+            with self.assertRaises(SystemExit):
+                self.invoke_httpd([tls_password_option, self.tls_password_file])
+                mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file cannot be opened
+
+        non_existent_file = os.path.join(tempfile.gettempdir(), os.urandom(16).hex())
+        retry_count = 0
+        while os.path.exists(non_existent_file) and retry_count < 10:
+            non_existent_file = os.path.join(tempfile.gettempdir(), os.urandom(16).hex())
+        if not os.path.exists(non_existent_file):
+            for tls_password_option in tls_password_options:
+                for tls_cert_option in tls_cert_options:
+                    with self.assertRaises(SystemExit):
+                        self.invoke_httpd([tls_cert_option, self.tls_cert, tls_password_option, non_existent_file])
+
+    def test_help_flag(self):
+        options = ['-h', '--help']
+        for option in options:
+            with self.assertRaises(SystemExit):
+                output = self.invoke_httpd([option])
+                self.assertIn('usage:', output)
```

Comment:
I'd prefer next method:
```suggestion
                self.assertStartsWith(output, 'usage: ')
```